gui/default/syncthing/core/syncthingController.js

```diff
@@ -1559,6 +1603,54 @@ angular.module('syncthing.core')
             return folderList($scope.folders);
         };
 
+        $scope.guiShowFolderList = function () {
+            var idleArray, pausedArray, syncingArray, unknownArray, stoppedArray, unsharedArray, defaultArray = [];
+
+            // loop through all folders
+            var folders = $scope.folderList();
+            for (var i = 0; i < folders.length; i++) {
+                var status = $scope.folderStatus(folders[i]);
+
+                switch (status){
+                case 'idle':
+                    idleArray.push(folders[i]);
+                    break;
+
+                case 'paused':
+                    pausedArray.push(folders[i]);
+                    break;
+
+                case 'syncing' || 'scanning':
+                        syncingArray.push(folders[i]);
+                    break;
+
+                case 'unknown':
+                    unknownArray.push(folders[i]);
+                    break;
+
+                case 'stopped' || 'outofsync' || 'error':
+                    stoppedArray.push(folders[i]);
+                    break;
+
+                case 'unshared':
+                    unsharedArray.push(folders[i]);
+                    break;
+
+                default:
+                    defaultArray.push(folders[i]);
+                }
+            };
+
+            idleArray = folderList(idleArray)
+            pausedArray = folderList(pausedArray)
+            syncingArray = folderList(syncingArray)
+            unknownArray = folderList(unknownArray)
+            stoppedArray = folderList(stoppedArray)
+            unsharedArray = folderList(unsharedArray)
+            defaultArray = folderList(defaultArray)
```

Comment:
Wouldn't it be better to concat the Arrays first and invoke `folderList()`only once?

Reply:
if we concat arrays and than call `folderList` we turn back default sorting base on `id` so `name`.